eth/src/main/kotlin/com/d3/eth/provider/EthTokensProviderImpl.kt

```diff
@@ -61,14 +82,19 @@ class EthTokensProviderImpl(
     override fun getTokenAddress(assetId: String): Result<String, Exception> {
         return if (assetId == "$ETH_NAME#$ETH_DOMAIN")
             Result.of { ETH_ADDRESS }
-        else getAccountDetails(
-            queryAPI,
-            tokenStorageAccount,
-            tokenSetterAccount
-        ).map { tokens ->
-            tokens.filterValues {
-                it == assetId
-            }.keys.first()
+        else irohaQueryHelper.getAccountDetails(
+            ethAnchoredTokenStorageAccount,
+            ethAnchoredTokenSetterAccount
+        ).fanout {
+            irohaQueryHelper.getAccountDetails(
+                irohaAnchoredTokenStorageAccount,
+                irohaAnchoredTokenSetterAccount
+            )
+        }.map { (ethAnchored, irohaAnchored) ->
+            val res = ethAnchored.plus(irohaAnchored).filterValues { it == assetId }
+            if (res.isEmpty())
+                throw IllegalArgumentException("Token $assetId not found")
+            res.keys.first()
```

Comment:
`first()` will throw an exception in case of empty collection, I think we can just 
`return ethAnchored.plus(irohaAnchored).filterValues { it == assetId }.keys.first()`
here